common/archiver/README.md

```diff
@@ -17,32 +18,59 @@ This README explains how to create new implementations for the History and Visib
       - visibilityArchiver.go       -- VisibilityArchiver implementations
       - visibilityArchiver_test.go  -- Unit tests for VisibilityArchiver
 ```
-2. Each Archiver will be given a `BootstrapContainer` when it's created and the archiver implementation can (will) use them to read history, emit logs/metrics etc. See `interface.go` for more details.
 
-3. Implement the Archiver interface. Both Archiver interfaces contain three methods:
-  
+**Step 2: Implement the HistoryArchiver interface**
+
 ```go
-func Archive(ctx context.Context, URI string, request *ArchiveRequest, opts ...ArchiveOption) error
+type HistoryArchiver interface {
+	// Archive is used to archive a workflow history. When the context expires the method should stop trying to archive.
+	// Implementors are free to archive however they want, including implementing retries of sub-operations. The URI defines
+	// the resource that histories should be archived into. The implementor gets to determine how to interpret the URI.
+	// The Archive method may or may not be automatically retried by the caller. The ArchiveOptions are used
+	// to interact with these retries including giving the implementor the ability to cancel retries and record progress
+	// between retry attempts. 
+    Archive(context.Context, URI, *ArchiveHistoryRequest, ...ArchiveOption) error
+    
+    // Get is used to access an archived history. When context expires method should stop trying to fetch history.
+    // The URI identifies the resource from which history should be accessed and it is up to the implementor to interpret this URI.
+    Get(context.Context, URI, *GetHistoryRequest) (*GetHistoryResponse, error)
```

Comment:
The `Get()` method is also responsible for casting the error the correct error type, e.g. BadRequest, EntityNotExist, etc.